### ISSUE TITLE
Support negation of requirements

### ISSUE THREAD

**Issue by Flamefire** [boostorg/build#610]
If you want to build something on Windows only my current solution was: `<build>no <target-os>windows:<build>yes` 

However this causes conflicts when any other rule (e.g. C++11 requirements) adds a `<build>no` too

Hence it would be great if negation like `<target-os>!windows:<build>no` would be supported

The current solution involves a rather clunky `conditional` and a function

**Comment by mloskot** [comment]
#594 is related as per https://lists.boost.org/Archives/boost/2020/05/248931.php

**Comment by Flamefire** [comment]
If I read this correctly this already exists as `!<target-os>windows:<build>no`. Is this correct?

**Comment by Kojoley** [comment]
Yes, but there seems to be another quirk that prevents to use negation on the first statement. I opened a PR with the error message reporting fix #611, and you are welcome to fix the bug itself.
